### PR TITLE
test(core): current mount pipeline

### PR DESCRIPTION
**Pull request by chirag-bruno**
### Description

[JIRA](https://usebruno.atlassian.net/browse/BRU-3240)

#### Contribution Checklist:

- [ ] **I've used AI significantly to create this pull request**
- [ ] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes**
- [ ] **I have added screenshots or gifs to help explain the change if applicable.**
- [ ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **Tests**
  - Expanded test coverage for collection mounting, loading state, sidebar tree structure, item ordering, and HTTP method indicators across multiple collection formats.
  - Added reusable fixtures covering various request methods and sequence scenarios.

* **Chores**
  - Added test utilities and page helpers to provision static fixtures and interact with the collection sidebar for more robust end-to-end validation.

### CODE REVIEW

File: tests/utils/page/mounting.ts [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+  /**
+   * Folder-specific locators
+   */
+  folder: {
+    /** Get folder row by name (exact match) */
+    row: (name: string) => page.getByTestId('sidebar-collection-item-row').filter({
+      has: page.locator('.item-name').getByText(name, { exact: true })
+    }).filter({
+      has: page.getByTestId('folder-chevron')
+    }),
```
Comment (sid-bruno) [inline tests/utils/page/mounting.ts]: To wide a boundary if you have 2 or more collections with the same folder name, this might pick the wrong one. You might wanna reuse/add a `openFolder` that takes the collection in consideration as well.

File: tests/utils/page/mounting.ts [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+    /** Get request row by name */
+    row: (name: string) => page.getByTestId('sidebar-collection-item-row').filter({
+      has: page.locator('.item-name', { hasText: name })
+    }).filter({
+      hasNot: page.getByTestId('folder-chevron')
+    }),
+    /** Get request method badge */
+    methodBadge: (name: string) => page.getByTestId('sidebar-collection-item-row').filter({
+      has: page.locator('.item-name', { hasText: name })
+    }).locator('.mr-1 span').first()
```
Comment (sid-bruno) [inline tests/utils/page/mounting.ts]: same problem as folder, this needs to be scoped to a collection/request row instead of globally searching for it in the sidebar

File: tests/utils/fixtures/generator.ts [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+ */
+export async function generateCollectionPair(
+  options: Omit<GenerateCollectionOptions, 'format'>
+): Promise<{
+  bru: GeneratedCollection;
+  yml: GeneratedCollection;
+  cleanup: () => Promise<void>;
+}> {
+  const bru = await generateCollection({ ...options, format: 'bru' });
+  const yml = await generateCollection({ ...options, format: 'yml' });
```
Comment (sid-bruno) [inline tests/utils/fixtures/generator.ts]: edge case but if the yml one throws , the bru one was still created and is left in the tmp dirs, can we just add a helper to make sure that if either fails and the `cleanup` function for the other is returned, you exec the cleanup. 

Something like this or if there's a better way then that

```suggestion
  const bru = await generateCollection({ ...options, format: 'bru' });
  const yml = await generateCollection({ ...options, format: 'yml' }).catch(err=>{
   ('cleanup' in bru ? bru.cleanup() : ()=>{})
   throw err;
  });
```

File: tests/utils/page/mounting.ts [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+  const locators = buildCollectionTreeLocators(page);
+
+  return await test.step(`Get tree structure for collection "${collectionName}"`, async () => {
+    const collectionRow = locators.collection.row(collectionName);
+
+    // Ensure collection is expanded
+    const isExpanded = await locators.collection.isExpanded(collectionName);
+    if (!isExpanded) {
+      await collectionRow.click();
+    }
```
Comment (sid-bruno) [inline tests/utils/page/mounting.ts:232]: Same as before comments, this could be anything in the generated set of collections if there's more than one

File: tests/mounting/loading-state.spec.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      });
+      await expect(collectionRow).toBeVisible({ timeout: 30000 });
+
+      // Loading spinner should not be visible
+      const loadingSpinner = collectionRow.locator('.animate-spin');
+      await expect(loadingSpinner).not.toBeVisible();
+
+      // Verify via helper
+      const loading = await isCollectionLoading(page, 'Test Collection');
+      expect(loading).toBe(false);
```
Comment (sid-bruno) [inline tests/mounting/loading-state.spec.ts]: The check for `.animate-spin` is redundant since `isCollectionLoading` does the same thing internally


```suggestion
      // Verify via helper
      const loading = await isCollectionLoading(page, 'Test Collection');
      expect(loading).toBe(false);
```

File: tests/mounting/tree-construction.spec.ts [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+      // we should have at least 2 different methods
+      expect(uniqueMethods.length).toBeGreaterThanOrEqual(2);
+
+      // All methods should be valid HTTP methods (UI truncates methods > 5 chars to 3 chars)
+      // DELETE -> DEL, OPTIONS -> OPT, etc.
+      const validMethods = ['GET', 'POST', 'PUT', 'DEL', 'PATCH', 'HEAD', 'OPT'];
+      for (const method of uniqueMethods) {
+        expect(validMethods).toContain(method);
+      }
+    });
```
Comment (sid-bruno) [inline tests/mounting/tree-construction.spec.ts]: There's nothing deterministic about this test, we are testing randomness and assuming the randomness might match. 

The assumption that it's greater than 2 is strictly based off the random algorithm that the generator has, i'd suggest we change this to an actual physical fixture instead of a generated one.

File: tests/utils/page/mounting.ts (hunk 4 above)
Comment (chirag-bruno) [reply tests/utils/page/mounting.ts:232]: `collectionRow` is already scoped by display name via `locators.collection.row(collectionName)` (`#sidebar-collection-name` filter), so this only collides if two mounted collections share the same name — not reachable in the current tests. Can add a disambiguator to `collection.row` when we actually mount duplicate-named collections.

File: tests/mounting/tree-construction.spec.ts (hunk 6 above)
Comment (chirag-bruno) [reply tests/mounting/tree-construction.spec.ts]: The generator isn't random — `generator.ts:86-87` cycles deterministically via `HTTP_METHODS[globalIndex % HTTP_METHODS.length]` over `[GET, POST, PUT, DELETE, PATCH]`, so with 10 requests every method always appears. The loose `>= 2` was just under-asserting. Tightened to `expect(new Set(methods)).toEqual(new Set(['GET','POST','PUT','DEL','PATCH']))` — deterministic and documents the contract, no physical fixture needed.

File: tests/utils/page/mounting.ts (hunk 4 above)
Comment (sid-bruno) [reply tests/utils/page/mounting.ts:232]: ignore, my bad

**Review by sid-bruno — APPROVED** [review]
